src/main/java/org/breedinginsight/brapps/importer/model/imports/PedigreeImportService.java

```diff
@@ -0,0 +1,489 @@
+/*
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.breedinginsight.brapps.importer.model.imports;
+
+import io.micronaut.http.server.exceptions.InternalServerException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.MutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.brapi.client.v2.model.exceptions.ApiException;
+import org.brapi.client.v2.modules.germplasm.CrossesApi;
+import org.brapi.v2.model.core.BrAPIProgram;
+import org.brapi.v2.model.germ.*;
+import org.brapi.v2.model.pheno.BrAPIObservationUnit;
+import org.breedinginsight.brapps.importer.model.base.*;
+import org.breedinginsight.brapps.importer.model.config.ImportRelation;
+import org.breedinginsight.brapps.importer.model.config.ImportRelationType;
+import org.breedinginsight.brapps.importer.services.BrAPIFileService;
+import org.breedinginsight.brapps.importer.services.BrAPIQueryService;
+import org.breedinginsight.model.Program;
+import org.breedinginsight.services.brapi.BrAPIClientType;
+import org.breedinginsight.services.brapi.BrAPIProvider;
+import org.breedinginsight.services.exceptions.UnprocessableEntityException;
+import tech.tablesaw.api.Table;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Singleton
+public class PedigreeImportService extends BrAPIImportService {
+
+    private String ID = "PedigreeImport";
+
+    private BrAPIQueryService brAPIQueryService;
+    private BrAPIFileService brAPIFileService;
+    private BrAPIProvider brAPIProvider;
+
+    @Inject
+    public PedigreeImportService(BrAPIQueryService brAPIQueryService, BrAPIFileService brAPIFileService, BrAPIProvider brAPIProvider)
+    {
+        this.brAPIQueryService = brAPIQueryService;
+        this.brAPIFileService = brAPIFileService;
+        this.brAPIProvider = brAPIProvider;
+    }
+
+    @Override
+    public PedigreeImport getImportClass() {
+        return new PedigreeImport();
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public BrAPIPreviewResponse process(List<BrAPIImport> brAPIImports, Table data, Program program, Boolean commit) throws UnprocessableEntityException {
+
```

Comment:
I think we can reduce this to just create `Germplasm` and `Crosses`. I thought I needed Observation units as well to track the relation of individuals to populations. But I see that the `Germplasm` model has a `collection` field where the population name can go.

Reply:
Chris and I talked about this and it will more closely match how imports work through breedbase as well.